docs/userguide/campaigns.md

```diff
@@ -63,7 +63,7 @@ experiments to be conducted.
 
 ~~~python
 rec = campaign.recommend(batch_size=3)
-print(rec.to_markdown())
```

Comment:
why?

Reply:
See commit message: `to_markdown` actually requires pandas to be installed with the optional `tabulate` dependency, which we don't have in our core environment. That means:
* The core tests fail (don't ask me why they passed prior to this PR)
* A user with core environment only would not be able to run this code. In general, I don't think this is a requirement, but specifically for the `campaign` user guide it's bad, I think.

So either we
* kick the `to_markdown` (which I did here)
* or add the dependency to the core deps via pandas enxtension (which I wouldn't want to do)
* or we skip the test case in `core`, which neither easy to do (since part of a for loop) nor would that solve the user issue

What do you prefer?

Reply:
sus that it didnt fail before... was the additional dependency provided by another requirement anyway and has now gone?

Depending on how the output looks im OK with this, but I think the normal df print potentially look bad?

Reply:
Ok, I got curious and search for the reason. Took me a bit but found the culprit: `tabulate` is installed through `pydoclint`, which comes from our `lint` group that is installed as part of the `test` group. That is, if you have an actual `core` environment (IMPORTANT difference: not the one we use in CI, because that one also installs `test`) containing no optional groups at all, you cannot run the code from the user guide – which is definitely bad.

So this indicates two things:
* Our "core" test isn't really a "core" test because it installs deps that should not be there. The only easy way around it that I currently see is to skip all tests that require linting in the core tests (otherwise, we'd need to build an isolated environment while running the tests, which is not so straightfoward). What is your opinion here, shall I try to implement the necessary changes?
* Once the above is done, our "core" pipeline should also **correctly** fail for the `to_markdown` call, i.e. we would see the problems in CI and no longer just by chance how I did when I ran locally in a pure core env

Finally, note that the `to_markdown` call is unnecessary anyway: the userguide is not "executed" in CI, hence we don't see the output! It's rather just an artifact from old times and would have vanished in the next overhaul. So I can just remove the entire line instead of just the call.

Reply:
Hey @Scienfitz, did exactly the above, that is:
* Remove the "lint" group from "test"
* Skipped the format tests in "core", which then failed
* Together, this revealed the failing test case of missing `tabulate`
* Removed to print statement completely to fix it

Please have a look at the last few commits and let me know if you agree / simply resolve this thread. Then I can merge 🚀

Reply:
ty for clarifying,t he way you handled it makes absolute sense, resolving